MulticastDeserializationTest test failure

- Fails on `Hazelcast-4.maintenance-ZuluJDK15`
- Fails on [Build #3 (Nov 8, 2020 7:30:42 AM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.maintenance-ZuluJDK15/3/testReport/junit/com.hazelcast.cluster/MulticastDeserializationTest/test/)
- Error
```
Untrusted deserialization is possible
```
- Stacktrace
```
java.lang.AssertionError: Untrusted deserialization is possible
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertFalse(Assert.java:65)
	at com.hazelcast.cluster.MulticastDeserializationTest.test(MulticastDeserializationTest.java:94)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:832)

```

- Noticed
```
09:35:07,829  WARN |testWithoutFilter| - [MulticastService] hz.nifty_bartik.MulticastThread - [127.0.0.1]:5701 [dev] [4.0.4-SNAPSHOT] class example.serialization.TestDeserialized cannot be cast to class com.hazelcast.internal.cluster.impl.JoinMessage (example.serialization.TestDeserialized and com.hazelcast.internal.cluster.impl.JoinMessage are in unnamed module of loader 'app')
java.lang.ClassCastException: class example.serialization.TestDeserialized cannot be cast to class com.hazelcast.internal.cluster.impl.JoinMessage (example.serialization.TestDeserialized and com.hazelcast.internal.cluster.impl.JoinMessage are in unnamed module of loader 'app')
	at com.hazelcast.internal.cluster.impl.MulticastService.receive(MulticastService.java:249) [classes/:?]
	at com.hazelcast.internal.cluster.impl.MulticastService.run(MulticastService.java:200) [classes/:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
```
- kindly check

--- a/hazelcast/src/test/java/com/hazelcast/cluster/MulticastDeserializationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cluster/MulticastDeserializationTest.java
@@ -74,8 +74,8 @@ public class MulticastDeserializationTest {
 
     @AfterClass
     public static void cleanup() {
-        TestDeserialized.isDeserialized = false;
         HazelcastInstanceFactory.terminateAll();
+        TestDeserialized.isDeserialized = false;
     }
 
     /**

